### PR TITLE
common/table-printer: separate columns with tab in non-TTY stdout

### PR DESCRIPTION
**Pull request by da2x**
Makes table output parsable for scripting.

### CODE REVIEW

**Comment by mwleeds** [comment]
This sort of implies that the output will be stable across flatpak releases, which it hasn't been so far.

**Comment by mwleeds** [comment]
Also `flatpak list | awk '{print $2}'` works for me, as long as none of the fields contain a space (which I think should be true?)

**Comment by da2x** [comment]
> This sort of implies that the output will be stable across flatpak releases, which it hasn't been so far.

All I’m asking for here is that columns are separated with the character expected by tools like `cut` and `ls`. I’m not asking for stable output between releases, although that would be nice.

> "Also flatpak list | awk '{print $2}' works for me, as long as none of the fields contain a space (which I think should be true?)"

Add a repository with a title containing a space. If you then awk the output of `flatpak remotes --show-details`, you’d have a problem with space separated columns. Or try to `cut` after you add one repository with a six character name and then add another with a ten character name.

**Comment by mwleeds** [comment]
bot, test pull request

File: common/flatpak-table-printer.c [hunk 1]
```diff
@@ -226,10 +226,17 @@ flatpak_table_printer_print (FlatpakTablePrinter *printer)
       for (j = 0; j < row->len; j++)
         {
           Cell *cell = g_ptr_array_index (row, j);
-          if (cell->align < 0)
-            g_print ("%s%-*s", (j == 0) ? "" : " ", widths[j], cell->text);
+          if (flatpak_fancy_output ())
+          {
```
Comment (mwleeds) [inline common/flatpak-table-printer.c]: Indentation error

Comment (da2x) [reply common/flatpak-table-printer.c]: Squashed in an indentation fix.

File: common/flatpak-table-printer.c [hunk 2]
```diff
@@ -226,10 +226,17 @@ flatpak_table_printer_print (FlatpakTablePrinter *printer)
       for (j = 0; j < row->len; j++)
         {
           Cell *cell = g_ptr_array_index (row, j);
-          if (cell->align < 0)
-            g_print ("%s%-*s", (j == 0) ? "" : " ", widths[j], cell->text);
+          if (flatpak_fancy_output ())
+            {
+            if (cell->align < 0)
```
Comment (mwleeds) [inline common/flatpak-table-printer.c]: This and all the lines added below it also need indentation fixes to match the convention in the rest of the codebase.

Comment (da2x) [reply common/flatpak-table-printer.c]: I’ve actually never seen this particular code style before so I got a bit confused. (No sleep probably had something to do with it too.) I think I got it right this time.

**Comment by mwleeds** [comment]
@rh-atomic-bot r+ fe15cc2

**Pull request closed without merging**